include/sound/hda_codec.h

```diff
@@ -253,6 +253,7 @@ struct hda_codec {
 	unsigned int force_pin_prefix:1; /* Add location prefix */
 	unsigned int link_down_at_suspend:1; /* link down at runtime suspend */
 	unsigned int relaxed_resume:1;	/* don't resume forcibly for jack */
+	unsigned int mst_no_extra_pcms:1; /* no backup PCMs for DP-MST */
```

Comment:
@ranj063 This is a feature Libin added many years back to the legacy driver. Main links here:
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=a76056f2e57ec2825d6ae4be016596dd37300750
https://www.kernel.org/doc/html/v4.10/sound/hd-audio/dp-mst.html

... we keep using the dynamic PCM assignment logic added back then. The difference is whether we allocate extra PCMs (e.g. 2 extras in case of current i915 + 5 PCMs in total) or not (limit to 3PCMs as that's the maximum number of converters).